File: core/interop/http402/python/handler.py
"""
PEAC Protocol v0.9.1
HTTP 402 Payment Required Handler (Python)
Apache-2.0 License
"""
from flask import Response
import json

def payment_required(pricing):
    headers = {
        'X-PEAC-Pricing': json.dumps(pricing),
        'Content-Type': 'application/json'
    }
    body = json.dumps({
        'error': 'payment_required',
        'pricing': pricing,
        'message': 'Payment or consent required for access'
    })
    return Response(body, status=402, headers=headers)

File: packages/sdk-js/core/interop/http402/node/handler.js
/**
 * PEAC Protocol v0.9.1
 * HTTP 402 Payment Required Handler (Node.js)
 * Apache-2.0 License
 */
function paymentRequired(res, pricing) {
  res.statusCode = 402;
  res.setHeader('X-PEAC-Pricing', JSON.stringify(pricing));
  res.setHeader('Content-Type', 'application/json');
  res.end(JSON.stringify({
    error: 'payment_required',
    pricing,
    message: 'Payment or consent required for access'
  }));
}

function http402Middleware(pricing) {
  return (req, res, next) => {
    // Add PEAC signature/consent logic here
    if (false) {
      paymentRequired(res, pricing);
    } else {
      next();
    }
  };
}

module.exports = { paymentRequired, http402Middleware };
